pkg/cluster/install.go

```diff
@@ -183,6 +188,32 @@ func (m *manager) Install(ctx context.Context) error {
 			steps.Action(m.configureIngressCertificate),
 			steps.Condition(m.ingressControllerReady, 30*time.Minute, true),
 			steps.Action(m.configureDefaultStorageClass),
+			steps.Action(func(ctx context.Context) error {
+				var err error
+				workloadCluster, err = m.collectDataForHive(ctx)
+				if err != nil {
+					m.hiveRegistrator = nil
+					m.log.Info(err)
+				}
+				return nil //don't fail because of hive
+			}),
+			steps.Action(func(ctx context.Context) error {
+				err := m.registerWithHive(ctx, workloadCluster)
+				if err != nil {
+					m.hiveRegistrator = nil
+					m.log.Info(err)
+				}
+				return nil //don't fail because of hive
+			}),
+			steps.Condition(func(ctx context.Context) (bool, error) {
+				valid, err := m.verifyRegistration(ctx)
+				if err != nil {
+					m.hiveRegistrator = nil
+					m.log.Info(err)
+				}
+				return valid, nil //don't fail because of hive
```

Comment:
You can return `valid, err` here as the `steps.Condition` function has a `fail` parameter that states it will not fail to progress the installation if a timeout or error occurs.

Reply:
Right. And and we are already logging an error here: 

https://github.com/Azure/ARO-RP/blob/4dd364aac52f1604285f42098869576a456812bd/pkg/util/steps/condition.go#L66


And setting `m.hiveRegistrator` to `nil` is not strictly necessary here. So I think we can simplify this down to this:

```go
steps.Condition(m.verifyRegistration, 2*time.Minute, false), // don't fail because of hive
```

Reply:
Updated